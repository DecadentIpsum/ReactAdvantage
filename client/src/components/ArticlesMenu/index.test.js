import React from 'react';
import Enzyme, { shallow } from 'enzyme';

import ArticleMenu from './index';

it('renders correctly', () => {
    const component = shallow(<ArticleMenu />);

    expect(component).toMatchSnapshot();
});

describe('popup is showing after click on button', () => {
    const component = shallow(<ArticleMenu />);
    const button = component.find('.artciles-menu-button');

    test('popup is hidden by default', () => {
        expect(component.state().visible).toEqual(false);
    });

    test('popup is visible after click on trigger button', () => {
        button.simulate('click');
        expect(component.state().visible).toEqual(true);
    });
});
