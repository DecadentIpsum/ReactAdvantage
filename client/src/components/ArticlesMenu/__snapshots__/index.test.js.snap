// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SidebarMenu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="artciles-menu-button"
          onClick={[Function]}
>
          <i
                    className="fa far fa-list-alt"
          />
</button>,
        Object {
          "$$typeof": Symbol(react.portal),
          "children": <Sidebar
            className="articles-sidebar"
            onHide={[Function]}
            position="right"
            visible={false}
>
            <div
                        className="articles-wrapper"
            >
                        <h2
                                    className="sidebar-title"
                        >
                                    What's new?
                        </h2>
                        <Article
                                    article={Object {}}
                                    index={0}
                        />
                        <Article
                                    article={Object {}}
                                    index={1}
                        />
                        <Article
                                    article={Object {}}
                                    index={2}
                        />
                        <Article
                                    article={Object {}}
                                    index={3}
                        />
            </div>
</Sidebar>,
          "containerInfo": <div />,
          "implementation": null,
          "key": null,
        },
      ],
      "className": "artciles-menu-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fa far fa-list-alt"
/>,
          "className": "artciles-menu-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa far fa-list-alt",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "button",
      },
      Object {
        "$$typeof": Symbol(react.portal),
        "children": <Sidebar
          className="articles-sidebar"
          onHide={[Function]}
          position="right"
          visible={false}
>
          <div
                    className="articles-wrapper"
          >
                    <h2
                              className="sidebar-title"
                    >
                              What's new?
                    </h2>
                    <Article
                              article={Object {}}
                              index={0}
                    />
                    <Article
                              article={Object {}}
                              index={1}
                    />
                    <Article
                              article={Object {}}
                              index={2}
                    />
                    <Article
                              article={Object {}}
                              index={3}
                    />
          </div>
</Sidebar>,
        "containerInfo": <div />,
        "implementation": null,
        "key": null,
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="artciles-menu-button"
            onClick={[Function]}
>
            <i
                        className="fa far fa-list-alt"
            />
</button>,
          Object {
            "$$typeof": Symbol(react.portal),
            "children": <Sidebar
              className="articles-sidebar"
              onHide={[Function]}
              position="right"
              visible={false}
>
              <div
                            className="articles-wrapper"
              >
                            <h2
                                          className="sidebar-title"
                            >
                                          What's new?
                            </h2>
                            <Article
                                          article={Object {}}
                                          index={0}
                            />
                            <Article
                                          article={Object {}}
                                          index={1}
                            />
                            <Article
                                          article={Object {}}
                                          index={2}
                            />
                            <Article
                                          article={Object {}}
                                          index={3}
                            />
              </div>
</Sidebar>,
            "containerInfo": <div />,
            "implementation": null,
            "key": null,
          },
        ],
        "className": "artciles-menu-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fa far fa-list-alt"
/>,
            "className": "artciles-menu-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fa far fa-list-alt",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "button",
        },
        Object {
          "$$typeof": Symbol(react.portal),
          "children": <Sidebar
            className="articles-sidebar"
            onHide={[Function]}
            position="right"
            visible={false}
>
            <div
                        className="articles-wrapper"
            >
                        <h2
                                    className="sidebar-title"
                        >
                                    What's new?
                        </h2>
                        <Article
                                    article={Object {}}
                                    index={0}
                        />
                        <Article
                                    article={Object {}}
                                    index={1}
                        />
                        <Article
                                    article={Object {}}
                                    index={2}
                        />
                        <Article
                                    article={Object {}}
                                    index={3}
                        />
            </div>
</Sidebar>,
          "containerInfo": <div />,
          "implementation": null,
          "key": null,
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
